### PR TITLE
Application manifest

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix MANAGER-5859
| License          | BSD 3-Clause

## Description

Add `@ovh-ux/manager-manifest-generator`: 

```sh
$ yarn run manager-manifest-generator --help
Usage: manager-manifest-generator [options] <basePath>

Options:
  -V, --version                output the version number
  --application <application>  application name (default: "application")
  -f, --file <filename>        manifest file name (default: "manifest.json")
  --verbose                    output extra debugging
  -h, --help                   display help for command
```

- Add manifest generation in `hub` application

## Related 
- [ ] https://github.com/ovh/manager/pull/4005 — `url-resolver`

## TODO
- [ ] add manifest generation in all apps (if the review is ok )

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#16960.0 ✘  
* 
  * Build ✘

File: packages/manager/apps/hub/package.json [hunk 1]
```diff
@@ -11,9 +11,10 @@
   "license": "BSD-3-Clause",
   "author": "OVH SAS",
   "scripts": {
-    "build": "webpack --env.production",
+    "build": "webpack --env.production && yarn run generate:manifest",
```
Comment (marie-j) [inline packages/manager/apps/hub/package.json:14]: No need to generate in  dev mode ?

File: yarn.lock [hunk 2]
```diff
@@ -1812,10 +1812,10 @@
     globals "^11.1.0"
     lodash "^4.17.19"
 
-"@babel/types@^7.0.0", "@babel/types@^7.10.5", "@babel/types@^7.12.1", "@babel/types@^7.12.5":
-  version "7.12.6"
-  resolved "https://registry.yarnpkg.com/@babel/types/-/types-7.12.6.tgz#ae0e55ef1cce1fbc881cd26f8234eb3e657edc96"
-  integrity sha512-hwyjw6GvjBLiyy3W0YQf0Z5Zf4NpYejUnKFcfcUhZCSffoBBp30w6wP2Wn6pk31jMYZvcOrB/1b7cGXvEoKogA==
+"@babel/types@^7.0.0", "@babel/types@^7.12.1":
+  version "7.12.1"
```
Comment (marie-j) [inline yarn.lock:1816]: Strange that the version has been downgraded here but I guess it's automatic with used dependencies

File: packages/manager/apps/hub/package.json (hunk 1 above)
Comment (cbourgois) [reply packages/manager/apps/hub/package.json:14]: As it parse the `dist` it won't work in dev.

File: yarn.lock (hunk 2 above)
Comment (cbourgois) [reply yarn.lock:1816]: Will check.

**Comment by cbourgois** [comment]
Hello reviewers ( @marie-j, @antleblanc & @frenautvh  )

I'm wondering if we should name our manifest `manifest.json`. 
`manifest.json` seems to be the name commonly used for PWA manifests (the name is not reserved but I would like to avoid conflicts / misunderstanding). 
What do you think ?

**Comment by cbourgois** [comment]
> Hello reviewers ( @marie-j, @antleblanc & @frenautvh )
> 
> I'm wondering if we should name our manifest `manifest.json`.
> `manifest.json` seems to be the name commonly used for PWA manifests (the name is not reserved but I would like to avoid conflicts / misunderstanding).
> What do you think ?

Following https://www.w3.org/TR/appmanifest/#using-a-link-element-to-link-to-a-manifest
> The official file extension for the manifest is `.webmanifest`.

So no issue.

**Pull request closed without merging**